src/ota_imxrt.c

```diff
@@ -258,7 +270,7 @@ MG_IRAM static bool mg_imxrt_erase(void *addr) {
   MG_ARM_DISABLE_IRQ();
   ret = (flexspi_nor_get_config(&config) == 0);
   if (ret) ret = flash_erase(&config, addr);
-  MG_ARM_ENABLE_IRQ();
+  if (!s_flash_irq_disabled) MG_ARM_ENABLE_IRQ();
```

Comment:
No, this one disabled through `#if 0` is intended to be a standalone call to just erase flash, it is not meant to be called in the OTA process itself, just if we reinstall our data load/save schema in the future.
As it works by itself, it needs to reenable IRQs when exiting

Reply:
OTA code only calls `flash_erase()`, not this function